clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetricsContext.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.metrics;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A implementation of MetricsContext, it encapsulates required metrics context properties for Kafka
+ */
+public class KafkaMetricsContext implements MetricsContext {
+    public static final String METRICS_CONTEXT_PREFIX = "metrics.context.";
+
+    /**
+     * Client or Service's metadata map.
+     */
+    private Map<String, String> metadata = new HashMap<>();
+
+    public KafkaMetricsContext() {
+    }
+
+    public KafkaMetricsContext(String namespace) {
+        metadata.put(MetricsContext.NAMESPACE, namespace);
```

Comment:
rather than repeat this code in both constructors, it's usually preferred to have only one constructor do all the initialization and have the overloaded ones call the main one with default arguments.